_adr/87/index.adoc

```diff
@@ -15,31 +15,92 @@ applies_patterns: # What APs does this ADR apply?
+
+* https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByInstantQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query]: returns the most recent values for each metric associated with the specified Kafka.
+This returns metrics in a JSON format.
+* https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByRangeQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query_range]: returns values for each metric associated with the specified Kafka over a time range specified in the request.
+This returns metrics in a JSON format.
+* https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/federateMetrics[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/federate]: returns the most recent values for each metric associated with the specified Kafka.
+This returns metrics in a Prometheus Text format and is scrapable by any monitoring systems that accepts this format.
+
+Each of these endpoints returns the exact same metrics.
+The only difference between these is the format of the response.
+
+The metrics returned in these endpoints are what we call customer facing metrics.
+All available customer facing metrics are documented in the app-services-guides: https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka]. 
+
+We currently expose two different types of customer facing metrics:
+
+* Raw Kafka metrics -- exposed by Kafka instances.
+* Recording rules -- configured by us for any Kafka metrics requiring pre-computed aggregation.
+
+This results in multiple problems:
+
+* *Consistency*: These two types of metrics follow different naming schemes, resulting in inconsistent metric names.
+* *Repetition*: The raw Kafka metrics contain a lot of repetition i.e. `kafka_controller_kafkacontroller...`.
+* *Exposure of implementation details*: The name of the recording rules metrics exposes implementation details i.e. `kafka_namespace:haproxy_server_bytes_in_total:rate5m`, end users do not need to know about namespaces and OpenShift.
+
+Since we don't have a formal process and naming policy for adding customer facing metrics, we often reach a situation where the resulting metric names are confusing and misleading (i.e. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/issues/688[misleading metric name for kafka topic partitions count]).
 
 ## Goals
-// Bulleted list of outcomes that this ADR, if accepted, should help achieve
+- Define a naming policy for existing and new customer facing metrics.
+- Define a process for requesting new customer facing metrics to the API.
+- Define a process for migrating the existing customer facing metrics to follow the new naming policy.
 
 ## Non-goals
-// Bulleted list of outcomes that this ADR is not trying to achieve.
+- Propose solution for metrics API versioning.
```

Comment:
After thinking about it, it may be best for us not to include this in the ADR but take it into consideration as part of the whole KAS Fleet Manager API versioning strategy. This has also been brought up a lot in recent discussions within the Control Plane team.